scaleway/resource_compute_instance_security_group.go

```diff
@@ -0,0 +1,465 @@
+package scaleway
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayComputeInstanceSecurityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceSecurityGroupCreate,
+		Read:   resourceScalewayComputeInstanceSecurityGroupRead,
+		Update: resourceScalewayComputeInstanceSecurityGroupUpdate,
+		Delete: resourceScalewayComputeInstanceSecurityGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the security group",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the security group",
+			},
+			"inbound_default_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "accept",
+				Description: "Default inbound traffic policy for this security group",
+				ValidateFunc: validation.StringInSlice([]string{
+					instance.SecurityGroupPolicyAccept.String(),
+					instance.SecurityGroupPolicyDrop.String(),
+				}, false),
+			},
+			"outbound_default_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "accept",
+				Description: "Default outbound traffic policy for this security group",
+				ValidateFunc: validation.StringInSlice([]string{
+					instance.SecurityGroupPolicyAccept.String(),
+					instance.SecurityGroupPolicyDrop.String(),
+				}, false),
+			},
+			"inbound_rule": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "inbound rules for this security group",
+				Elem:        securityGroupRuleSchema(),
+			},
+			"outbound_rule": {
```

Comment:
Maybe prular name, since it is a list?

Reply:
Same as `inbound_rule`

Reply:
```hcl
resource "scaleway_compute_instance_security_group" "base" {
  inbound_rule {
      ...
  }

  outbound_rule {
      ...
  }
}
```

As it will be used like this in the config file, the singular is better IMO.